publishable/database/migrations/2017_11_26_013050_add_user_role_relationship.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddUserRoleRelationship extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->unsignedInteger('role_id')->nullable(false)->change();
+            $table->foreign('role_id')->references('id')->on('roles');
```

Comment:
We didn't already have a foreign key here...? huh...

Reply:
We did not, but if we allow nullables, we should remove this foreign key?

Reply:
MySQL (at least) allows null values in a foreign key field (so long as it's not a UNIQUE index): https://stackoverflow.com/questions/7573590/can-a-foreign-key-be-null-and-or-duplicate.

I think the foreign key is fine

Reply:
fixed